yt_dlp/extractor/slideslive.py

```diff
@@ -473,7 +532,18 @@ def _real_extract(self, url):
         if service_name == 'url':
             info['url'] = service_id
         elif service_name == 'yoda':
-            info['formats'] = self._extract_formats(player_info['video_servers'][0], service_id, video_id)
+            formats, duration = self._extract_formats_and_duration(
+                player_info['video_servers'][0], service_id, video_id)
+            if not duration:
+                last_start_time = traverse_obj(
+                    info, ('chapters', -1, 'start_time'), expected_type=int_or_none)
+                # if chapters, prevent core code from setting the last chapter's end_time as None
+                if last_start_time:
+                    info['chapters'][-1]['end_time'] = last_start_time
```

Comment:
This is wrong? Can you explain?

Reply:
When chapter duration/end_time is not known and video duration cannot be extracted, this piece of core code will set the last chapter's end_time as `None` (`info_dict.get('duration')`)
https://github.com/yt-dlp/yt-dlp/blob/87ebab0615b1bf9b14b478b055e7059d630b4833/yt_dlp/YoutubeDL.py#L2519-L2531

This breaks embedding chapters:
https://github.com/yt-dlp/yt-dlp/blob/87ebab0615b1bf9b14b478b055e7059d630b4833/yt_dlp/postprocessor/ffmpeg.py#L713-L728

It also breaks splitting chapters:
https://github.com/yt-dlp/yt-dlp/blob/87ebab0615b1bf9b14b478b055e7059d630b4833/yt_dlp/postprocessor/ffmpeg.py#L1039-L1040

To prevent this breakage, I'm setting the only known safe value for the last chapter's `end_time`: its `start_time`, since the `start_time` could potentially already be the end of the video. I don't like it much myself, but it's being used as a fallback to a fallback. Is there a smarter way to fix this in the core/postprocessing code, do you think?

Reply:
The issue affects all sites and so I don't think this makes sense to do in the extractor. We have a few options:
1. Do what you did, but in core code
2. At postprocessing stage, we have access to the downloaded video, and so could determine the actual duration with ffprobe (`self._get_real_video_duration`)
3. What happens if we add the `END=` line only when `end_time` is known? Afaik, it should mark the chapter till end of file. Similar for `-t`

Whichever we go for, this is out of scope for this PR imo. You can revert just this part and merge the rest for now

Reply:
Did some testing, and `#3` will work for splitting chapters, but it doesn't really work for embedding chapters (at least mpv will not detect the last chapter or does not consider it valid).

I prefer `#2` to what I did. So we could do `#2` for `FFmpegMetadataPP` and do `#3` for `FFmpegSplitChaptersPP`?

This PR should be good to go now btw